src/test/java/com/exadel/movie_new/tests/MovieServiceTest.java

```diff
@@ -0,0 +1,138 @@
+        newMovie2.setGenres(List.of("Comedy", "Romance"));
+    }
+
+
+
+    @Test
+    @DisplayName("Should throw exception that movie already exists")
+    void shouldThrowExceptionMovieAlreadyExists() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieAlreadyExists.class, () -> movieService.addMovie(newMovie));
+    }
+
+    @Test
+    @DisplayName("Should fetch all movies")
+    void shouldGetAllMovies() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertFalse(movies.isEmpty());
+        assertEquals(2, movies.size());
+    }
+
+    @Test
+    @DisplayName("Should return an empty list")
+    void shouldGetEmptyList() {
+        List<Movie> list = new ArrayList<>();
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertTrue(movies.isEmpty());
+        assertEquals(0,movies.size());
+    }
+
+    @Test
+    @DisplayName("Should fetch movie by Id")
+    void shouldGetMovieById() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+
+        when(movieDaoImplMock.read()).thenReturn(list);
+        Movie existingMovie = movieService.getMovie(newMovie.getId());
+        assertNotNull(existingMovie);
+        assertNotEquals(null, existingMovie.getId());
+        assertEquals(newMovie.getId(), existingMovie.getId());
+    }
+
+    @Test
+    @DisplayName("Should throw MovieNotFoundException while getMovie")
+    void shouldGetMovieByIdException() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieNotFoundException.class, () -> movieService.getMovie("200"));
+    }
+
+    @Test
+    @DisplayName("Should thrown MovieNotFoundException while updateMovie")
+    void shouldUpdateMovieException() {
+        List<Movie> list = new ArrayList<>();
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieNotFoundException.class, () -> movieService.updateMovie(newMovie));
+    }
+
+    @Test
+    @DisplayName("Should delete movie")
+    void shouldDeleteMovie() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+
+        List<Movie> listAfterDelete = new ArrayList<>();
+        listAfterDelete.add(newMovie2);
+
+        String movieId = "100";
+        when(movieDaoImplMock.read()).thenReturn(list);
+        movieService.deleteMovie(movieId);
+        assertEquals(1, list.size());
+        verify(movieDaoImplMock, timeout(1)).write(listAfterDelete);

```

Comment:
I'm not sure why you need this timeout(1). You can leave it for now, because it does not brake the logic but it looks like redundant